src/inmanta/ast/statements/assign.py

```diff
@@ -317,7 +325,16 @@ def requires_emit(self, resolver: Resolver, queue: QueueScheduler) -> typing.Dic
     def resume(
         self, requires: typing.Dict[object, object], resolver: Resolver, queue: QueueScheduler, target: ResultVariable
     ) -> None:
-        self.type.lookup_index([(k, v.execute(requires, resolver, queue)) for (k, v) in self.query], self, target)
+        self.type.lookup_index(
```

Comment:
no error is raise upon duplicate arguments.

It seems that this was not in place in the past either, but we should check it either way

I think it should be added to the lookup_index method itself

Reply:
two testcases you should add
```python
def test_index_lookup(snippetcompiler):
    snippetcompiler.setup_for_snippet(
        """
entity Test:
    string a
end

index Test(a)

implement Test using std::none

b = Test(a="b")
a = Test(a="a")

ar = Test[a="a"]
"""
    )
    (_, scopes) = compiler.do_compile()
    root = scopes.get_child("__config__")
    a = root.lookup("a").get_value()
    ar = root.lookup("ar").get_value()
    assert a == ar


def test_index_lookup_double_argument(snippetcompiler):
    snippetcompiler.setup_for_snippet(
        """
entity Test:
    string a
end

index Test(a)

implement Test using std::none

b = Test(a="b")
a = Test(a="a")

ar = Test[a="a", a="b"]
"""
    )
    compiler.do_compile()
```

Reply:
(second one needs to be adapted to the way you make it fail )

Reply:
The first test case is just a general index lookup test. Such a test does not exist yet?
Also, I've used 'is' instead of '==' in assertions in the other index tests I've written. What's your opinion on this?

Reply:
1. I didn't find it
2. no opinion 

@bart: any opnion?

Reply:
well, is is identity match and == equality. Indexes should return the same object and not a similar object. So that means is
in this case it does not matter that much because we do not override __eq__
but who knows what happens in the future